src/components/pdf-downloader/PDFDownloader.jsx

```diff
@@ -0,0 +1,32 @@
+import { saveAs } from 'file-saver'
+
+export const downloadPDF = async (pdfDocument, pdfUrl) => {
+  const pdfName = pdfUrl.substring(pdfUrl.lastIndexOf('/') + 1)
+
+  try {
+    // Try to download file via blob to avoid opening PDF in another PDF plugin
+    const pdfData = await pdfDocument.getData()
+    const blob = new Blob([pdfData], { type: 'application/pdf' })
+    saveAs(blob, pdfName)
+  } catch (error) {
+    // If blob is not supported or something happens
+    // download it via traditional way
+    // Code taken from https://github.com/mozilla/pdf.js/blob/master/web/download_manager.js#L28
+    const a = document.createElement('a')
+    a.href = pdfUrl
+    a.target = '_blank'
+
+    // Use a.download if available. This increases the likelihood that
+    // the file is downloaded instead of opened by another PDF plugin.
+    if ('download' in a) a.download = pdfName
+
+    // <a> must be in the document for IE and recent Firefox versions,
+    // otherwise .click() is ignored.
+    const rootDocument = document.body || document.documentElement
+    rootDocument.appendChild(a)
+    a.click()
```

Comment:
Check the [function from `file-saver`](https://github.com/eligrey/FileSaver.js/blob/master/src/FileSaver.js#L58-L67) by the way

Reply:
By the way, if the button that triggers this function already is an anchor with download attribute and correct name, you do not need to have to catch anything. 

Also, you might not need this at all because the document probably is cached after the first network request. However, if think about downloading annotated document, there should be an event listener.